mlx_graphs/datasets/lanl_dataset.py

```diff
@@ -0,0 +1,320 @@
+    _preprocessed_drive_id = "11rSBKagmOzmC-Xru4mUyTfep6P7FGmcO"
+
+    def __init__(
+        self,
+        process_original_files: bool = False,
+        use_gzip: bool = True,
+        num_nodes: int = LANL_NUM_NODES,
+        base_dir: Optional[str] = None,
+        **kwargs,
+    ):
+        self.auth_file = "auth.txt.gz"
+        self.redteam_file = "redteam.txt.gz"
+        self.tmp_archive_file = "LANL.zip"
+
+        self._use_gzip = use_gzip
+        self.process_original_files = process_original_files
+
+        extension = "csv.gz" if use_gzip else "csv"
+        super().__init__(
+            name="LANL",
+            raw_file_extension=extension,
+            num_nodes=num_nodes,
+            base_dir=base_dir,
+            **kwargs,
+        )
+
+    @property
+    def original_path(self) -> str:
+        """
+        We call original files, the files that are downloaded from the official
+        LANL website. We only use 2 of these files: "auth.txt.gz" and "redteam.txt.gz".
+
+        Returns:
+            The path to the folder that contains original files.
+        """
+        root = "/".join(self.raw_path.split("/")[:-1])
+        return os.path.join(root, "original")
+
+    def download(self):
+        """
+        By default, downloads the preprocessed form of the LANL dataset.
+        This version comprises a "LANL.zip" file of size ~340MB, containing
+        a csv.gz snapshot file for each minute of data.
+
+        If ``process_original_files=True``, the original "auth.txt.gz" and
+        "redteam.txt.gz" files are downloaded and preprocessed to re generate
+        the files in "LANL.zip".
+        """
+        if not self.process_original_files:
+            download_file_from_google_drive(
+                id=self._preprocessed_drive_id,
+                path=os.path.join(self.raw_path, self.tmp_archive_file),
+            )
+            extract_archive(
+                os.path.join(self.raw_path, self.tmp_archive_file), self.raw_path
+            )
+
+            # "LANL" is the name of the folder after decompression
+            extracted_folder = os.path.join(self.raw_path, "LANL")
+            files = os.listdir(extracted_folder)
+
+            for file in files:
+                source_file = os.path.join(extracted_folder, file)
+                destination_file = os.path.join(self.raw_path, file)
+
+                shutil.move(source_file, destination_file)
+
+            shutil.rmtree(extracted_folder, ignore_errors=True)
+
+        else:
+            # NOTE: This downloads the 7.6GB auth.txt.gz file from LANL.
+            # It is extremly long to dl due to LANL's slow servers (up to 2hr)
+            if not os.path.exists(os.path.join(self.original_path, self.auth_file)):
+                download(
+                    url=os.path.join(self._url, self.auth_file),
+                    path=os.path.join(self.original_path, self.auth_file),
+                )
```

Comment:
Is this accessible without authentication?

Reply:
Yes, there's just two fields of a form to fill on the website but there is no login requirement to fetch the files directly from the URL.